enumerable.rb

```diff
@@ -0,0 +1,145 @@
+# frozen_string_literal: true
+# rubocop:disable all
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+    i = 0
+      while i < size
+        yield self[i]
+        i += 1
+      end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+    i = 0
+      while i < size
+        yield(self[i], i)
+        i += 1
+      end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+    array2 = []
+      i = 0
+      while i < size
+        array2 << self[i] if yield self[i]
+        i += 1
+      end
+    array2
+  end
+
+  def my_all?(pattern = nil)
+    test = true
+    if(block_given?)
+      self.my_each do |x|
+        test = test && yield(x)
+      end 
+    else
+      if(pattern.nil?)
+        self.my_each do |x|
+          test = test && (x==true)
+        end
+      else
+        self.my_each do |x|
+          test = test && (x.match?(pattern))
+        end
+      end
+    end
+    return test
+  end
+
+  def my_any?(pattern = nil)
+    test = false
+    if(block_given?)
```

Comment:
`[nil, true, 99].any?(Integer)                     #=> true`
- [ ] my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class